Support `root` boot parameter (#3425)
asterinas/asterinas: book/src/kernel/linux-compatibility/kernel-parameters.md

## Inherited from Linux

### `rdinit`

Run the specified initramfs binary as the first userspace process.

Example:
```text
rdinit=/bin/busybox
```

Notes:
- The value is the path to the executable in the initramfs root.
- If omitted, Asterinas will try to execute `/init` from the initramfs root.
- If the specified path cannot be accessed,
  Asterinas ignores it and proceeds with root filesystem initialization.
  It does not fall back to `/init` in this case.

### `root`

Mount the specified block device as the root filesystem
when no initramfs init is selected to run.

Example:
```text
root=/dev/vda2
```

Notes:
- The value currently must name a registered block device under `/dev`,
  such as `/dev/vda2`.
- By default, the initramfs init is `/init`;

>   `rdinit` overrides it with another path.

Reviewer: This should be put under the `rdinit` section?

- If the selected initramfs init is present,

>   that program is responsible for switching to the root filesystem.

Reviewer: I cannot understand line 38-39. It seems to be the userspace behavior? Should we include it here?

Or should we reword it as "If rdinit is provided, root= is ignored...."

### `rootfstype`

Select the filesystem type used for the root filesystem.

Example:
```text
root=/dev/vda2 rootfstype=ext2
```

Valid values:
- `ext2`

### `ro`

Mount the root filesystem read-only.
This is the default when Asterinas mounts the root filesystem via `root=`.

Example:
```text
root=/dev/vda2 ro
```

Notes:

> - This parameter is ignored when an initramfs init is selected to run.

Reviewer: ```suggestion
- This parameter is ignored when `rdinit=` is provided, or when `root=` is not provided.
```

Would this be more accurate? By the way, do parameters like `ro`/`rw` need to immediately follow `root=`, or can other parameters appear in between?

Author: Sorry for the confusion. By “an initramfs init is selected,” I mean that the kernel chooses the initramfs path rather than the rootfs path. In this case, `ro` and `rw` have no effect.  I will expand the note to make “selected” explicit.

“When root= is not provided” is not an ignored case if no initramfs init was selected—the kernel will panic because it has neither an initramfs init nor a root filesystem to boot from.

`ro` and `rw` are parsed independently and do not need to immediately follow `root=`. Other parameters may appear between them, and if both are specified, the last one takes precedence.
